include/core/Layer.hpp

```diff
@@ -0,0 +1,147 @@
+            std::string current_timestamp = simulation_time.get_timestamp(output_time_index);
+            for(const auto& id : processing_units) 
+            {
+                int sub_time = output_time_index;
+                //std::cout<<"Running cat "<<id<<std::endl;
+                auto r = features.catchment_at(id);
+                //TODO redesign to avoid this cast
+                auto r_c = std::dynamic_pointer_cast<realization::Catchment_Formulation>(r);
+                r_c->set_et_params(pdm_et_data);
+                double response = r_c->get_response(output_time_index, description.time_step);
+                std::string output = std::to_string(output_time_index)+","+current_timestamp+","+
+                                    r_c->get_output_line_for_timestep(output_time_index)+"\n";
+                r_c->write_output(output);
+                //TODO put this somewhere else.  For now, just trying to ensure we get m^3/s into nexus output
+                try{
+                    response *= (catchment_data->get_feature(id)->get_property("areasqkm").as_real_number() * 1000000);
+                }catch(std::invalid_argument &e)
+                {
+                    response *= (catchment_data->get_feature(id)->get_property("area_sqkm").as_real_number() * 1000000);
+                }
+                //TODO put this somewhere else as well, for now, an implicit assumption is that a modules get_response returns
+                //m/timestep
+                //since we are operating on a 1 hour (3600s) dt, we need to scale the output appropriately
+                //so no response is m^2/hr...m^2/hr * 1hr/3600s = m^3/hr
+                response /= 3600.0;
+                //update the nexus with this flow
+                for(auto& nexus : features.destination_nexuses(id)) {
+                    //TODO in a DENDRIDIC network, only one destination nexus per catchment
+                    //If there is more than one, some form of catchment partitioning will be required.
+                    //for now, only contribute to the first one in the list
+                    nexus->add_upstream_flow(response, id, output_time_index);
+                    std::cerr << "Add water to nexus ID = " << nexus->get_id() << " from catchment ID = " << id << " value = "
+                              << response << ", ID = " << id << ", time-index = " << output_time_index << std::endl; 
+                    break;
+                }
+                
+            } //done catchments   
+
+            ++output_time_index;
+            simulation_time.advance_timestep();       
+        }
+        
+
+        protected:
+
+        LayerDescription description;
+        std::vector<std::string> processing_units;
+        Simulation_Time simulation_time;
+        feature_type& features;
+        geojson::GeoJSON catchment_data;
+        std::shared_ptr<pdm03_struct> pdm_et_data;
+        long output_time_index;       
```

Comment:
Whichever of these members can be `const` ought to be.

Reply:
Not many of them can be constant although in most cases the accessors could be. Catchment_data probably could be but I would have to check to see if the GeoJSON class works correctly with a const pointer.